conan/tools/meson/toolchain.py

```
@@ -168,22 +168,27 @@ def __init__(self, conanfile, backend=None, native=False):
         # Values are kept as Python built-ins so users can modify them more easily, and they are
         # only converted to Meson file syntax for rendering
         # priority: first user conf, then recipe, last one is default "ninja"
-        self._backend = self._conanfile_conf.get("tools.meson.mesontoolchain:backend",
-                                                 default=backend or 'ninja')
+        #: Backend to use. Defined by the conf ``tools.meson.mesontoolchain:backend``. By default, ``ninja``.
+        self.backend = self._conanfile_conf.get("tools.meson.mesontoolchain:backend",
```

But shouldn't the value be both configurable in recipe and overridable by ``conf``. This way, users that put something like ``tc.backend = "..."`` will be hardcoding the value, and when they try to define ``tools.meson.mesontoolchain:backend=xxxx`` in a profile or command line, it will be silently ignored for that recipe and that can be confusing.

It looks like (I think the pattern is similar in other generators):
- the ``__init__()`` defines some initial default value
- the user can define with ``tc.xxxx = yyyy`` some deviation from those defaults
- when really assigning the values, if the conf like ``tools.meson.mesontoolchain:backend`` is defined, it will be used?

---

Oh, you're completely right.